six/modules/c++/cphd/source/CPHDWriter.cpp

```diff
@@ -32,14 +32,20 @@ namespace cphd
 CPHDWriter::DataWriter::DataWriter(io::FileOutputStream& stream,
             size_t numThreads) :
     mStream(stream),
-    mNumThreads(numThreads)
+    mNumThreads(initNumThreads(numThreads))
 {
 }
 
 CPHDWriter::DataWriter::~DataWriter()
 {
 }
 
+size_t CPHDWriter::initNumThreads(size_t numThreads)
```

Comment:
This function is simple enough and won't get reused anywhere that I think it's actually simpler just to do this right in the initializer list so the reader doesn't have to go find this function.